docs/services/ml/upload-training-script.md

```diff
@@ -0,0 +1,216 @@
+# specifically for getting the bounding box annotations
+def parse_filenames_and_bboxes_from_json(
+    filename: str,
+    all_labels: ty.List[str],
+) -> ty.Tuple[ty.List[str], ty.List[str], ty.List[ty.List[float]]]:
+    """Load and parse JSON file to return image filenames
+    and corresponding labels with bboxes.
+    Args:
+        filename: JSONLines file containing filenames and bboxes
+    """
+    image_filenames = []
+    bbox_labels = []
+    bbox_coords = []
+
+    with open(filename, "rb") as f:
+        for line in f:
+            json_line = json.loads(line)
+            image_filenames.append(json_line["image_path"])
+            annotations = json_line["bounding_box_annotations"]
+            labels = []
+            coords = []
+            for annotation in annotations:
+                if annotation["annotation_label"] in all_labels:
+                    labels.append(annotation["annotation_label"])
+                    # Store coordinates in rel_yxyx format so that
+                    # we can use the keras_cv function
+                    coords.append(
+                        [
+                            annotation["y_min_normalized"],
+                            annotation["x_min_normalized"],
+                            annotation["y_max_normalized"],
+                            annotation["x_max_normalized"],
+                        ]
+                    )
+            bbox_labels.append(labels)
+            bbox_coords.append(coords)
+    return image_filenames, bbox_labels, bbox_coords
+```
+
+{{% /expand%}}
+{{%expand "Step 3: Add logic to produce model artifact" %}}
+
+After reading in the dataset and preparing the data for training, you can add logic to build and compile the model and train the model on the data.
+The code in this part of your script depends on the type of ML training you are doing.
+
+{{% /expand%}}
+{{%expand "Step 4: Save the model artifact the script produces" %}}
+
+You also need to include in your <file>training.py</file> logic to save the model artifact your training job produces to the `model_output_directory`.
+For example:
+
+```python
+# This allows us to save the model artifact to Viam,
+# which will be viewable as a registry item for the ML model name
+# and version specified
+def save_tflite_classification(
+    model: Model,
+    model_dir: str,
+    model_name: str,
+    target_shape: ty.Tuple[int, int, int],
+) -> None:
+    # Convert the model to tflite, with batch size 1
+    # so the graph does not have dynamic-sized tensors.
+    input = tf.keras.Input(target_shape, batch_size=1, dtype=tf.uint8)
+    output = model(input, training=False)
+    wrapped_model = tf.keras.Model(inputs=input, outputs=output)
+    converter = tf.lite.TFLiteConverter.from_keras_model(wrapped_model)
+    converter.target_spec.supported_ops = TFLITE_OPS
+    tflite_model = converter.convert()
+
+    # Save the model to GCS
```

Comment:
This may be storing it to GCS in the implementation details but that's not something the user would have to know
```suggestion
```